### PR TITLE
12-error-handling is done

### PR DESCRIPTION
**Pull request by Hassen-Ahmed**

### CODE REVIEW

File: src/App.jsx [hunk 1]
```diff
@@ -18,6 +19,7 @@ function App() {
                 <Route path="/signin" element={<SignIn />} />
                 <Route path="/profile" element={<Profile />} />
                 <Route path="/articles/:article_id" element={<SingleArticle />} />
+                <Route path="*" element={<Error msg="Wrong url" />} />
```
Comment (popatre) [inline src/App.jsx:22]: Nice separate error component - nice and re-useable.

File: src/components/article/SingleArticle.jsx [hunk 2]
```diff
@@ -103,7 +111,11 @@ const SingleArticle = () => {
     return (
         <>
             {isLoading ? (
-                loading
+                !isWrongId ? (
+                    loading
+                ) : (
+                    <Error msg="  Article not Exist!" />
```
Comment (popatre) [inline src/components/article/SingleArticle.jsx]: I'd maybe just change this to **Article Does Not Exist**

**Review by popatre — APPROVED** [review]
- Nice 400 for your parametrics/queries that don't exist
- Good 404 for a path that doesnt exist 
- Good handling of a 404 for an article that doesn't exist - i'd maybe think of something a bit more user friends instead of `wrong query` - might be one to come back to later though
- Good work with the non-existent topic 404! 👍

File: src/components/article/SingleArticle.jsx (hunk 2 above)
Comment (Hassen-Ahmed) [reply src/components/article/SingleArticle.jsx]: Thank you so much! I will fix that.

**Pull request merged**